contrib/babelfishpg_tsql/antlr/TSqlParser.g4

```diff
@@ -3956,16 +3955,34 @@ geospatial_func_arg
     | STINTERSECTS
     ;
 
+geospatial_static_method
+    : STGEOMFROMTEXT
+    | STPOINTFROMTEXT
+    | POINT
+    ;
+
 hierarchyid_methods
     : method=( GETANCESTOR | GETDESCENDANT | GETLEVEL | ISDESCENDANTOF | READ | GETREPARENTEDVALUE | TOSTRING ) LR_BRACKET expression_list? RR_BRACKET
     ;
 
-hierarchyid_coloncolon_methods
-    : id colon_colon  method=(GETROOT | PARSE) LR_BRACKET expression? RR_BRACKET
+hierarchyid_static_method
+    : GETROOT
+    | PARSE
+    ;
+
+datatype_static_method
+    : hierarchyid_static_method
+    | geospatial_static_method
+    ;
+
+datatype_field_or_method
+    : method=id (LR_BRACKET expression_list? RR_BRACKET)
```

Comment:
which test case will qualify/match with this rule?

Reply:
For methods and fields that does not fall under `datatype_static_method` will hit this rule 
```
select geometry::NonExistentMethod()
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Could not find method 'NonExistentMethod' for type geometry.)~~


select geography::NonExistentMethod()
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Could not find method 'NonExistentMethod' for type geography.)~~


select xml::NonExistentMethod()
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Cannot call methods on xml.)~~
```

Reply:
```
select geography::NonExistentMethodField
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Could not find property or field 'NonExistentMethodField' for type geography.)~~


select xml::NonExistentMethodField
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Cannot call methods on xml.)~~


select int::NonExistentMethodField
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Cannot call methods on int.)~~

```
For more tests refer: https://github.com/rohit01010/babelfish_extensions/blob/d0590aeb089c4cb7caa6f166909112e429a79a84/test/JDBC/expected/colon_colon_methods-vu-verify.out#L225